cmd/tsbs_generate_data/common/distribution_test.go

```diff
@@ -96,3 +96,50 @@ func TestFloatPrecisionGet(t *testing.T) {
 		}
 	}
 }
+
+func TestSaddledDistribution(t *testing.T) {
+	testCases := []struct {
+		desc      string
+		threshold float64
+		saddle    Distribution
+		step      Distribution
+		expect    float64
+	}{
+		{
+			desc:      "threshold not satisfied, value remains the same",
+			threshold: 1,
+			saddle:    &mockDistribution{ReturnValue: 0},
+			step:      CWD(&mockDistribution{ReturnValue: 1}, 2, 4, 2),
+			expect:    float64(2),
```

Comment:
don't think you need to cast this, I think the compiler infers it given the type of the field